### PR TITLE
Clean up diffs in Trainer/TFTrainer

### PR DESCRIPTION
**Pull request by sgugger**
This PR does a bit of cleanup in the two Trainer and tries ti make the diff in the two TrainingArguments as minimal as possible.

- `set_seed` is now just one function in trainer_utils: the problem was that even if you only use TF and import it from transformers right now, it does not set seed for tf **and** will fail on PyTorch stuff.
- `eval_steps` is now a common argument for both versions of Trainer
- as discussed, `n_gpu` in `TFTrainingArguments` becomes `n_replicas`. This is a breaking change, I can add the deprecation warnings that goes with it if you think it's necessary.

### CODE REVIEW

File: src/transformers/training_args.py [hunk 1]
```diff
@@ -102,6 +102,8 @@ class TrainingArguments:
         dataloader_drop_last (:obj:`bool`, `optional`, defaults to :obj:`False`):
             Whether to drop the last incomplete batch (if the length of the dataset is not divisible by the batch size)
             or not.
+        eval_steps (:obj:`int`, `optional`, defaults to 1000):
+            Number of update steps before two evaluations.
```
Comment (LysandreJik) [inline src/transformers/training_args.py:106]:
```suggestion
            Number of update steps between two evaluations.
```

**Review by LysandreJik — COMMENTED** [review]
Looks good! I would indeed add a property `n_gpu` returning `n_replicas` with a `FutureWarning`, which would prevent a breaking change.

File: src/transformers/training_args_tf.py [hunk 2]
```diff
@@ -85,18 +85,17 @@ class TFTrainingArguments(TrainingArguments):
...
             Wheter to activate the trace to record computation graphs and profiling information or not.
     """
 
     tpu_name: str = field(
         default=None, metadata={"help": "Name of TPU"},
     )
-    eval_steps: int = field(default=1000, metadata={"help": "Run an evaluation every X steps."})
     debug: bool = field(
         default=False, metadata={"help": "Activate the trace to record computation graphs and profiling information"}
     )
```
Comment (jplu) [inline src/transformers/training_args_tf.py:101]: What about merging this parameter with `tpu_metrics_debug`?

**Review by jplu — APPROVED** [review]
LGTM! Just few minor comments

**Comment by julien-c** [comment]
Pretty cool 🔥

File: src/transformers/training_args_tf.py [hunk 3]
```diff
@@ -85,18 +85,17 @@ class TFTrainingArguments(TrainingArguments):
         dataloader_drop_last (:obj:`bool`, `optional`, defaults to :obj:`False`):
             Whether to drop the last incomplete batch (if the length of the dataset is not divisible by the batch size)
             or not.
-        tpu_name (:obj:`str`, `optional`):
-            The name of the TPU the process is running on.
         eval_steps (:obj:`int`, `optional`, defaults to 1000):
             Number of update steps before two evaluations.
```
[the remark this answers is not in the document]
Comment (sgugger) [reply src/transformers/training_args_tf.py:89]: If I remove it, it won't be documented anymore (the argument now comes from `TrainingArguments`).

[the remark this answers is not in the document]
Comment (jplu) [reply src/transformers/training_args_tf.py:89]: Oh indeed! Sorry for the wrong comment.

**Pull request merged**